ThirdPartyAdapters/chartboost/chartboost/src/main/java/com/google/ads/mediation/chartboost/ChartboostSingleton.java

```diff
@@ -208,27 +281,74 @@ private static void startChartboost(Activity activity, ChartboostParams params,
 
         if (!mIsChartboostInitialized) {
             mIsChartboostInitializing = true;
-            Chartboost.startWithAppId(activity, params.getAppId(), params.getAppSignature());
-
-            Chartboost.setMediation(Chartboost.CBMediation.CBMediationAdMob,
-                    BuildConfig.VERSION_NAME);
-            Chartboost.setLoggingLevel(CBLogging.Level.INTEGRATION);
-            Chartboost.setDelegate(getInstance());
-            Chartboost.setAutoCacheAds(false);
-
-            // Chartboost depends on Activity's lifecycle events to initialize its SDK. Chartboost
-            // requires onCreate, onStart and onResume callbacks to initialize its SDK.  By the time
-            // AdMob SDK requests this adapter to initialize the SDK, all three callbacks might have
-            // already been called. So, we call Chartboost's onCreate, onStart and onResume methods
-            // so that the Chartboost SDK will be initialized.
-            Chartboost.onCreate(activity);
-            Chartboost.onStart(activity);
-            Chartboost.onResume(activity);
+            initChartboostSdk(context, params);
+            initBannerWithListener(context, params, bannerListener);
```

Comment:
This really makes no sense to me. Can't we do it on `startChartboostBanner()`?
Doing this for interstitials is just weird, and requires you to go to the `initBannerWithListener` implementation to understand what is really going on.

Reply:
Fixed